### PR TITLE
Added BatchNorm class

### PR DESCRIPTION
**Pull request by HangenYuu**
- [x] Added BatchNorm class in `ivy/stateful/norms.py` based on the work at PR https://github.com/unifyai/ivy/pull/12779.
- [x] Did informal experiments at [this notebook](https://colab.research.google.com/drive/1yyx55LQxWQ8DNLMOyu3mxnpWRMMnUjUo?usp=sharing) to check adherence to the Torch module but receive questionable results that needs clarifying.
- [x] Ready to merge.

### CODE REVIEW

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by HangenYuu** [comment]
Hi [Arif](https://github.com/YushaArif99), appreciate it if you can take some time to review this and merge the branch if it is okay, or provide further comment so that I can fix accordingly. Thank you.

**Comment by YushaArif99** [comment]
Hi @HangenYuu 

Apologies for the late response. Was quite swamped with work recently and so reviewing this PR totally slipped under my radar 😅.

With regards to this PR, could you provide an issue no which this PR is referencing? The link you've shared seems to be related to a merge PR on refactoring `instance_norm`, `batch_norm`, and `layer_norm`. 

Is there some subtask that specifically mentions implementing `batch_norm` as a stateful module class?

**Comment by HangenYuu** [comment]
Hi @YushaArif99

I did this based on no subtask. I submitted a GSoC proposal on implementing models to contribute to Hugging Face Model Zoo, but I found that there is only `LayerNorm` stateful module class. As I will need `BatchNorm` for the proposal (if it is accepted that is 😅) I go on and implement it.

In that case, I have a feeling that this will fall under `Ivy API Experimental`?

**Review by YushaArif99 — COMMENTED** [review]
Hi @HangenYuu 

Apologies again for the delay in getting back to you 😓

Any updates regarding your proposal? I'd suggest you work on the open tasks that you can find on the [website](https://unify.ai/docs/ivy/overview/contributing/open_tasks.html#open-tasks) if the contribution is a part of the application process. 
But if you are willing to independently contribute as part of  GSOC, then it's best to ask this on [discord](https://discord.com/invite/G4aR9Q7DTN). Maybe Dan or someone from the IvyTeam can better guide you on how to proceed with this even without the official GSOC funding.

Let me know if you have any other questions 🙂
